### PR TITLE
[MM-10017] Added support to override right-hand sidebar

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary

- Implemented the RHS component which can be viewed when clicking the Channel Header Action Button.
- Updated Readme with component screenshot and description. 

#### Ticket Link
Fixes [MM-10017](https://github.com/mattermost/mattermost-server/issues/10169)

#### Related Pull Requests
- Includes changes in WebApp [PR #2827](https://github.com/mattermost/mattermost-webapp/pull/2827)

### CODE REVIEW

File: webapp/README.md [hunk 1]
```diff
@@ -119,4 +119,10 @@ This plugin registers a file upload action that, when clicked, triggers the root
 ## Link Tooltip
 
 This plugin registers a link tooltip component, whose content is fully customizable. It is displayed when the mouse cursor hovers over a link in a post.
-![link tooltip](docs/link_tooltip.png)
\ No newline at end of file
+![link tooltip](docs/link_tooltip.png)
+
+## Righ-Hand Sidebar
```
Comment (crspeller) [inline webapp/README.md]:
```suggestion
## Right-Hand Sidebar
```

File: webapp/src/plugin.jsx [hunk 2]
```diff
@@ -22,14 +24,15 @@ import {
...
     mainMenuAction,
-    channelHeaderButtonAction,
     fileUploadMethodAction,
     postDropdownMenuAction,
     websocketStatusChange,
     getStatus,
 } from './actions';
 import reducer from './reducer';
 
+const showPluginRHS = window.rhsPluginAction.showPluginRHS;
```
Comment (crspeller) [inline webapp/src/plugin.jsx]: See webapp pr.

**Comment by deanwhillier** [comment]
Apologies, but I'm not currently able to review this, adding @hmhealey as an alternate reviewer.

File: webapp/i18n/es.json [hunk 3]
```diff
@@ -4,6 +4,7 @@
   "plugin.upload": "Subir usando el plugin demo",
   "root.clicktoclose": "Haga clic en cualquier lugar para cerrar.",
   "root.triggered": "Has activado el componente raíz del complemento de demostración.",
+  "rhs.triggered": "Has activado el componente del panel derecho del complemento de demostración",
```
Comment (hmhealey) [inline webapp/i18n/es.json]: We might not want this file changed in this PR. Generally, PRs only change `en.json` and the other files are updated using the translation server. That may work differently for plugins though, so @crspeller would know better about it

Comment (crspeller) [reply webapp/i18n/es.json]: It's fine here. There isn't a translation process for plugins yet.

**Comment by crspeller** [comment]
Awaiting https://github.com/mattermost/mattermost-webapp/pull/2827

**Pull request merged**